src/org/ods/component/CopyImageOptions.groovy

```diff
@@ -0,0 +1,45 @@
+package org.ods.component
+
+import groovy.transform.TypeChecked
+
+@TypeChecked
+class CopyImageOptions extends Options {
+
+    /**
+     * Source image to import
+     *
+     * This needs to be in the following format: [REGISTRY/]REPO/IMAGE[:TAG]
+     */
+    String sourceImageUrlIncludingRegistry
+
+    /**
+     * true will tag the image from the -cd namespace into the targetEnvironment that the pipeline is running for
+     */
+    Boolean tagIntoTargetEnv
+
+    /**
+     * sourceCredential is the token to use, if any, to access the source registry
+     */
+    String sourceCredential
+
+    /**
+     * verifyTLS allows the stage to ignore certificate validation errors.
+     *
+     * The default is to verify certificate paths
+     */
+    Boolean verifyTLS
+
+    // INTERNAL!!
```

Comment:
Don't you want to use the `private` access modifier to make the internal nature transparent?

Reply:
@serverhorror You can remove the "X is an INTERNAL value" comments, since this is already indicated by the `private` access modifier. `protected` would have been an alternative if you need to make the same field available in a subclass, but that's not the case.